### PR TITLE
fix: wrong Ed25519 KID value

### LINKED ISSUE #2530: fix Ed25519 KID value
Currently KIDs are based on the public key's JWK thumbprint.

There's a [bug](https://github.com/square/go-jose/pull/324) for Ed25519 JWK thumbprints in go-jose.

Fix this manually in the framework as the go-jose fix might take a while to be merged in their Github repo.

### PR DESCRIPTION
**Pull request by Baha-sk**
This change fixes go-jose's Ed25519 JWK Thumbprint value used in the KMS KID creation.

It also includes a check for emptiness of the key bytes with calling CreateKID() and
fixed related unit tests in the framework.

Finally the webkms bdd tests are commented out since the remote kms server requires
an udpate. They will be uncommented in a future revision when the server image tag
is updated with a revision of aries-framework-go that contains this change.

closes #2530

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: pkg/doc/util/jwkkid/kid_creator.go [hunk 1]
```diff
@@ -69,11 +80,13 @@ func buildJWK(keyBytes []byte, kt kms.KeyType) (*jose.JWK, error) {
 		if err != nil {
 			return nil, fmt.Errorf("buildJWK: failed to build JWK from ecdsa DER key: %w", err)
 		}
-	case kms.ED25519Type:
-		jwk, err = jose.JWKFromPublicKey(ed25519.PublicKey(keyBytes))
-		if err != nil {
-			return nil, fmt.Errorf("buildJWK: failed to build JWK from ed25519 key: %w", err)
-		}
+		// TODO remove `case kms.ED25519Type` in CreateKID() and uncomment below case when go-jose fixes Ed25519
```
Comment (troyronda) [inline pkg/doc/util/jwkkid/kid_creator.go:83]: What is go Jose issue number

File: pkg/doc/util/jwkkid/kid_creator_test.go [hunk 2]
```diff
@@ -32,9 +36,23 @@ func TestCreateKID(t *testing.T) {
...
+	jwk, err := jose.JWKFromPublicKey(pubKey)
+	require.NoError(t, err)
+
+	goJoseTP, err := jwk.Thumbprint(crypto.SHA256)
+	require.NoError(t, err)
+
+	goJoseKID := base64.RawURLEncoding.EncodeToString(goJoseTP)
+	require.NotEqual(t, kid, goJoseKID)
+
+	// now build bad thumbprint and ensure it matches goJoseTP
```
Comment (troyronda) [inline pkg/doc/util/jwkkid/kid_creator_test.go:50]: Where is a test showing we are building the correct kid

File: pkg/doc/util/jwkkid/kid_creator.go (hunk 1 above)
Comment (Baha-sk) [reply pkg/doc/util/jwkkid/kid_creator.go:83]: there's no issue in go-jose repo, only a pending pr mentioned in the AFG issue #2530

File: pkg/doc/util/jwkkid/kid_creator_test.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: according to the pending pr, the fix is here https://github.com/square/go-jose/pull/324/files#diff-adb413cfea72e48e90449f9604b1472d63b22160f9c1230bb22517ece6a0fa89R334

In AFG, we build a correct JWK marshalled string then do the same as go-jose. I have not searched if there's an official source of JWK Thumbprint. Unfortunately IETF [RFC 7638](https://tools.ietf.org/html/rfc7638) has no mention of JWK's of Ed25519 keys and contains only an RSA example, so I take it go-jose's format is valid.

**Review by fqutishat — APPROVED** [review]

File: pkg/doc/util/jwkkid/kid_creator_test.go (hunk 2 above)
Comment (troyronda) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: We need a test that demonstrates the correct behavior, not just that it isn't the wrong one.

Comment (Baha-sk) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: added `TestCreateKIDFromFixedKey` further down at line 122

Comment (llorllale) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: * JWK thumbprint algo: https://tools.ietf.org/html/rfc7638#section-3
* JWK of public Ed25519 key: https://tools.ietf.org/html/rfc8037#appendix-A.2

**Review by llorllale — APPROVED** [review]

File: pkg/doc/util/jwkkid/kid_creator_test.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: I'll update the positive KID  test with https://tools.ietf.org/html/rfc8037#appendix-A.2 @llorllale , thanks

Comment (Baha-sk) [reply pkg/doc/util/jwkkid/kid_creator_test.go:50]: updated, Ed25519 JWK Thumbprint test now matches [rfc 8037 A.3](https://tools.ietf.org/html/rfc8037#appendix-A.3) example

**Pull request merged**